### PR TITLE
Reactor netty connection pooling

### PR DESCRIPTION
**Pull request by kushagraThapar**
* Removed explicit disposing connections after reading response. 
* Reactor netty automatically puts the connection back to connection pool after response is read: https://gitter.im/reactor/reactor-netty?at=5e11accfa769bf30fd3e8c0f
* Addresses this issue: https://github.com/Azure/azure-sdk-for-java/issues/6973
* Tested this PR with ReadThroughput benchmark and the results show that connections are being re-used
* Here are some snapshots when running readThroughput in Https and gateway mode. 
** Https mode: 
```
consistency-test@consistency-tests-machine:~/azure-sdk-for-java/sdk/cosmos$ netstat -ant | awk '{print $6}' | sort | uniq -c | sort -n
      1 Foreign
      1 established)
      3 ESTABLISHED
      3 LISTEN
consistency-test@consistency-tests-machine:~/azure-sdk-for-java/sdk/cosmos$ netstat -ant | awk '{print $6}' | sort | uniq -c | sort -n
      1 Foreign
      1 established)
      3 LISTEN
   2984 ESTABLISHED
consistency-test@consistency-tests-machine:~/azure-sdk-for-java/sdk/cosmos$ netstat -ant | awk '{print $6}' | sort | uniq -c | sort -n
      1 Foreign
      1 established)
      3 LISTEN
   3094 ESTABLISHED
consistency-test@consistency-tests-machine:~/azure-sdk-for-java/sdk/cosmos$ netstat -ant | awk '{print $6}' | sort | uniq -c | sort -n
      1 Foreign
      1 established)
      3 LISTEN
   3108 ESTABLISHED
consistency-test@consistency-tests-machine:~/azure-sdk-for-java/sdk/cosmos$ netstat -ant | awk '{print $6}' | sort | uniq -c | sort -n
      1 Foreign
      1 established)
      3 LISTEN
   3152 ESTABLISHED
consistency-test@consistency-tests-machine:~/azure-sdk-for-java/sdk/cosmos$ netstat -ant | awk '{print $6}' | sort | uniq -c | sort -n
      1 Foreign
      1 established)
      3 LISTEN
   3156 ESTABLISHED
consistency-test@consistency-tests-machine:~/azure-sdk-for-java/sdk/cosmos$ netstat -ant | awk '{print $6}' | sort | uniq -c | sort -n
      1 Foreign
      1 established)
      3 LISTEN
   3181 ESTABLISHED
consistency-test@consistency-tests-machine:~/azure-sdk-for-java/sdk/cosmos$ netstat -ant | awk '{print $6}' | sort | uniq -c | sort -n
      1 Foreign
      1 established)
      3 LISTEN
   3191 ESTABLISHED
consistency-test@consistency-tests-machine:~/azure-sdk-for-java/sdk/cosmos$ netstat -ant | awk '{print $6}' | sort | uniq -c | sort -n
      1 Foreign
      1 established)
      3 ESTABLISHED
      3 LISTEN
   3192 TIME_WAIT
```

** Gateway mode: 
```
consistency-test@consistency-tests-machine:~/azure-sdk-for-java/sdk/cosmos$ netstat -ant | awk '{print $6}' | sort | uniq -c | sort -n
      1 Foreign
      1 established)
      3 ESTABLISHED
      3 LISTEN
consistency-test@consistency-tests-machine:~/azure-sdk-for-java/sdk/cosmos$ netstat -ant | awk '{print $6}' | sort | uniq -c | sort -n
      1 Foreign
      1 established)
      3 LISTEN
    105 ESTABLISHED
consistency-test@consistency-tests-machine:~/azure-sdk-for-java/sdk/cosmos$ netstat -ant | awk '{print $6}' | sort | uniq -c | sort -n
      1 Foreign
      1 established)
      3 LISTEN
   1004 ESTABLISHED
consistency-test@consistency-tests-machine:~/azure-sdk-for-java/sdk/cosmos$ netstat -ant | awk '{print $6}' | sort | uniq -c | sort -n
      1 Foreign
      1 established)
      3 LISTEN
   1004 ESTABLISHED
consistency-test@consistency-tests-machine:~/azure-sdk-for-java/sdk/cosmos$ netstat -ant | awk '{print $6}' | sort | uniq -c | sort -n
      1 Foreign
      1 established)
      3 LISTEN
   1004 ESTABLISHED
consistency-test@consistency-tests-machine:~/azure-sdk-for-java/sdk/cosmos$ netstat -ant | awk '{print $6}' | sort | uniq -c | sort -n
      1 Foreign
      1 established)
      3 ESTABLISHED
      3 LISTEN
   1001 TIME_WAIT
```

### CODE REVIEW

File: sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/http/ReactorNettyClient.java [hunk 1]
```diff
@@ -172,31 +172,26 @@ public HttpHeaders headers() {
...
         }
 
         @Override
         public Mono<String> bodyAsString(Charset charset) {
-            return bodyIntern().aggregate().asString(charset).doFinally(s -> this.close());
+            return bodyIntern().aggregate().asString(charset);
         }
 
         @Override
         public void close() {
```
Comment (David-Noble-at-work) [inline sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/http/ReactorNettyClient.java]: Might the method override be removed now that the body of close is empty?

File: sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/http/ReactorNettyClient.java [hunk 2]
```diff
@@ -172,31 +172,26 @@ public HttpHeaders headers() {
...
-            if (reactorNettyConnection.channel().eventLoop().inEventLoop()) {
-                reactorNettyConnection.dispose();
-            } else {
-                reactorNettyConnection.channel().eventLoop().execute(reactorNettyConnection::dispose);
-            }
         }
```
Comment (David-Noble-at-work) [inline sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/http/ReactorNettyClient.java]: Might the method override be removed now that the body of close is empty?

Comment (kushagraThapar) [reply sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/http/ReactorNettyClient.java]: Makes sense, will remove the empty method.

Comment (kushagraThapar) [reply sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/http/ReactorNettyClient.java]: Done

**Pull request merged**